controllers/clusteroperator_test.go

```diff
@@ -0,0 +1,55 @@
+package controllers
+
+import (
+	"context"
+	"testing"
+
+	osconfigv1 "github.com/openshift/api/config/v1"
+	fakeconfigclientset "github.com/openshift/client-go/config/clientset/versioned/fake"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestUpdateCOStatusDisabled(t *testing.T) {
+	tCases := []struct {
+		name               string
+		expectedConditions []osconfigv1.ClusterOperatorStatusCondition
+		expected           bool
+	}{
+		{
+			name: "Incorrect Condition",
+			expectedConditions: []osconfigv1.ClusterOperatorStatusCondition{
+				setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, "", ""),
+				setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, "", ""),
+			},
+			expected: false,
+		},
+		{
+			name: "Correct Condition",
+			expectedConditions: []osconfigv1.ClusterOperatorStatusCondition{
+				setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, "", ""),
+				setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, "", ""),
+			},
+			expected: true,
+		},
+	}
+
+	reconciler := newFakeProvisioningReconciler(setUpSchemeForReconciler(), &osconfigv1.Infrastructure{})
+	co, _ := reconciler.createClusterOperator()
+	reconciler.OSClient = fakeconfigclientset.NewSimpleClientset(co)
+
+	for _, tc := range tCases {
+		reconciler.updateCOStatusDisabled()
+		gotCO, _ := reconciler.OSClient.ConfigV1().ClusterOperators().Get(context.Background(), clusterOperatorName, metav1.GetOptions{})
+
+		for _, expectedCondition := range tc.expectedConditions {
+			ok := v1helpers.IsStatusConditionPresentAndEqual(
+				gotCO.Status.Conditions, expectedCondition.Type, expectedCondition.Status,
+			)
+			if !ok {
+				assert.Regexp(t, tc.expected, ok)
+			}
+		}
+	}
+}
```

Comment:
We can keep updateCOStatusAvailable() in clusteroperator.go. Can we also add a test case to test the available state?

Reply:
removing available from this commit